draft-private-access-tokens.md

```diff
@@ -568,22 +645,45 @@ blind_sig = rsabssa_blind_sign(skP, AccessTokenRequest.blinded_req)
 
 `skP` is the private key corresponding to ORIGIN_TOKEN_KEY, known only to the Issuer.
 
+The Issuer also generates a value, "origin_tag", as follows:
+
+~~~
+origin_tag = HashToScalar(ORIGIN_NAME) * ISSUER_SECRET * request_tag
+~~~
+
 The Issuer generates an HTTP response with status code 200 whose body consists of
-blind_sig. The Issuer sends this as the response to the forwarded request,
-sets the media type to "message/access-token-response", and sends the result to
-the Mediator.
+blind_sig, with the content type set as "message/access-token-response" and the 
+origin_tag set in the "Origin-Tag" header.
+
+~~~
+:status = 200
+content-type = message/access-token-response
+content-length = 512
+origin-tag = origin_tag
+
+<Bytes containing the blind_sig>
+~~~
+
+### Mediator-to-Client Response {#response-two}
+
+Upon receipt of a successful response from the Issuer, the Mediator extracts the
+"Origin-Tag" header, and uses the value to determine ANON_ISSUER_ORIGIN_ID.
+
+~~~
+ANON_ISSUER_ORIGIN_ID = origin_tag / request_nonce
+~~~
 
-The Issuer then updates its local state for the ANON_CLIENT_ID, ensuring that the
-ORIGIN_NAME and ANON_ORIGIN_ID_PRIME values are stored if not already present, and
-the history of token issuance is recorded with sufficient granularity to apply the
-Issuer's policy. For example, if the policy is simply meant to bound the number of
-tokens given to a specific Client within the policy window, the Issuer needs to increment
-its counter of the the number of tokens issued for the ANON_CLIENT_ID.
+If the "Origin-Tag" is missing, or if the same ANON_ISSUER_ORIGIN_ID was previously
+received in a response for a different ANON_ORIGIN_ID within the same policy window,
```

Comment:
This search only needs to be against the ANON_ORIGIN_ID for this client, right? Otherwise needing to search against all seen ANON_ORIGIN_ID pairs might become expensive.